nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -545,7 +556,9 @@ def load_checkpoint(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
 
             # after dist_checkpointing.load, sharded tensors will be replaced with tensors
             checkpoint['state_dict'] = sharded_state_dict
-            checkpoint['optimizer_states'] = [self.optimizer_sharded_state_dict(is_loading=True)]
+            # Check whether to load optim states
+            if map_location.get("load_optim_states", True):
```

Comment:
We shouldn't use `map_location` for that purpose, it's not even in the base strategy interface?

For the record, this approach will work even if we don't restrict optimizer loading, because we won't save the optimizer state. But restricted load will be way faster because optimizer state is the majority of the checkpoint.

I would suggest adding a new argument to `load_checkpoint`, e.g. `load_optimizer_state: bool = True`